[Support Forum] Update the WP and JP app's support screen

guarani opened the pull request:
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/20116

This PR implements a few changes and updates to the support screen design. The changes are easiest to follow visually by looking at the screenshots below. 

The code changes are easiest to follow by looking at the commits:
1. 68b367c0c5a24aa4fbcbb38b1b22031ab056bfa8 removes the check for A8c emails from the WP app. This check is only relevant to Zendesk support which is no longer part of the WP app.
2. c2192dbb08a66082f017e962779afdcace1ce68e removes Zendesk code from an execution path specific to the WP app (no longer needed now that the WP app won't show Zendesk support).
3. 7964f4d1caa7bc49f41cf775173e8f054b2b249b remove the forum from the JP app (this wasn't clear in the issues I created, apologies).
4. 3d4c4329741c3e26b17c637758dddc213615154e incorporated PR review feedback (updated to the latest design, reamed & unified copy, and localized string keys)
5. 19cfbd90b168859e414b46b3fd04dd555d4620b9 removed resolved to-do code comment
6. 84850187f30ef8d758569f37dbdd099f041e3974 removed now unused code

(There are a few commit messages where I said "Zenhub" instead of "Zendesk" 😬 )

## Screenshots
| | Feature flag OFF  | Feature flag ON  |
| - | - | - |
| WordPress | <img width="539" alt="Screenshot 2023-02-13 at 20 54 39" src="https://user-images.githubusercontent.com/1898325/218604610-287d6281-90c2-47af-9575-6be5af5e4601.png"> | <img width="539" alt="Screenshot 2023-02-13 at 20 55 30" src="https://user-images.githubusercontent.com/1898325/218604622-376d0c9b-30e5-4fe8-ab73-2e0429530f1a.png"> |
| Jetpack | <img src="https://user-images.githubusercontent.com/1898325/218605384-cf443255-53ff-4041-8e18-76b50d1ad0c3.png" alt="" width="350"> | Feature flag should have no effect on Jetpack app  |

## Test

Please note: The feature flag mentioned below is the "Provide support through a forum" one.

### Ensure the JP app does not show the .org forum
1. Install the Jetpack app
2. Verify that the forum is **not** shown (the feature flag has no effect on the JP app)

### Ensure the A8c email popup is not shown
This can happen if the app has remembered an A8c email address as the user's support email
1. Install the WordPress app
2. Go to the support screen and make sure the A8c email popup doesn't show up
3. Verify that the forum is shown 

### Design & copy check
This PR also updated the support screen to the latest designs & copy (see qk5s4q4mZsXknwhMWo2vIY-fi-3882%3A14189). Please flag anything that looks off as the designs and copy have been tweaked recently.

## Out of scope of this PR
1. There's an ongoing discussion about what to do with "WordPress Help Center" (see p1675873769529629-slack-C3EA6SMFH)
2. There's also a discussion about when to ship this change while ensuring users wishing to contact support while trying to switch to the Jetpack app can do that (see p1676313024578649-slack-C04KJRDTLF7)
I'd like to leave these topics for future PRs if needed.


## Regression Notes
1. Potential unintended areas of impact

This change has the potential to inadvertently show the wrong support options in the WP and JP apps.

7. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing as seen in the above section

8. What automated tests I added (or what prevented me from doing so)

I looked at adding a test which instantiated `SupportTableViewController` and querying its view model to verify it has the right contents for each app (WP and JP). The trouble is that the view model contains private types (that conform to `ImmuTableRow`) so it's not possible to grab the view model's rows without making many types public (e.g. `HelpRow`). Happy to hear if there's a way to do it.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Hide .org forum on Jetpack

The JP app won't show the forum, only the WP app will

Commit:
Avoid check for A8c email on WP app

The check for an A8c email shows a popup that says "Please log in with a non-Automattic" email to submit or view support tickets" so this can be removed for the WP app (and kept only for the JP app).

Commit:
Remove dead code for Zenhub support in WP app

The WP app replaced Zenhub support with .org forum support, so the Zenhub-related code in the execution path unique to the the WP app was never going to be executed.

sla8c commented on WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift:
```diff
@@ -138,24 +141,6 @@ private extension SupportTableViewController {
 
     func tableViewModel() -> ImmuTable {
```
With these changes in this PR I think it would be clearer if we rename this function name to be `wpTableViewModel()`?  And also consequentially as a result should `oldTableViewModel()` be renamed to `jpTableViewModel()` and the `TODO` be removed?

guarani replied:
Good idea! I'll update this.

sla8c commented:
@guarani thanks for this PR!  Sorry small nit but it would be nice if the test steps had with and without feature lag outcome / expectations since this is departing from the figma design and I'm not sure what should be the expected the result with and without feature flags cc @staskus

sla8c commented:
@Guarani also with the changes to no longer user FeatureFlag in SupportViewController I think it would be best to remove the TODO on line 513 in SupportViewController and rename the localizedstring keys to better reflect Jetpack vs WordPress thanks!

Commit:
Update support screen to latest design

guarani commented:
> Sorry small nit but it would be nice if the test steps had with and without feature lag outcome / expectations since this is departing from the figma design and I'm not sure what should be the expected the result with and without feature flags

Thanks for suggesting that! I added screenshots for cases (feature flag on/off) and for both apps.

> @guarani also with the changes to no longer user FeatureFlag in SupportViewController I think it would be best to remove the TODO on line 513 in SupportViewController and rename the localizedstring keys to better reflect Jetpack vs WordPress thanks!

I've updated this as well. I added a new test case to cover checking the design and copy. 

This is ready to review again 🙇

sla8c commented on WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift:
```diff
@@ -197,47 +182,47 @@ private extension SupportTableViewController {
         return ImmuTable(sections: sections)
     }
 
     // TODO - remove after FeatureFlag.wordPressSupportForum is removed
```
@guarani I think we can remove this TODO as well?

guarani replied:
Missed that one, thanks @sla8c! I pushed the change to remove this.

Commit:
Removed resolved TODO note

sla8c commented on WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift:
```diff
@@ -17,6 +17,7 @@ class SupportTableViewController: UITableViewController {
     private var tableHandler: ImmuTableViewHandler?
     private let userDefaults = UserPersistentStoreFactory.instance()
     private let featureFlagStore: RemoteFeatureFlagStore
```
@guarani since we are changing this class to no longer be dependent on FeatureFlag I think we can remove this property as well?  Since SupportConfiguration has its own default parameter for FeatureFlag there should be no impact

guarani replied:
Good call! I didn't see this, will update.

sla8c commented on WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift:
```diff
@@ -197,47 +182,46 @@ private extension SupportTableViewController {
         // Help Section
         var helpSectionRows = [ImmuTableRow]()
-        helpSectionRows.append(HelpRow(title: LocalizedText.wpHelpCenter, action: helpCenterSelected(), accessibilityIdentifier: "help-center-link-button", featureFlagSupportForum: featureFlagStore.value(for: FeatureFlag.wordPressSupportForum)))
+        helpSectionRows.append(HelpRow(title: LocalizedText.wpHelpCenter, action: helpCenterSelected(), accessibilityIdentifier: "help-center-link-button", featureFlagSupportForum: isForumShown))
```
@guarani parameter `featureFlagSupportForum` for `HelpRow` no longer makes sense here since this parameter was used to control the text color depending on feature flag state.  Thanks for those screenshots on before and after feature flag; since the colour is no longer changing based on feature flag this is redundant now.  Thanks!

guarani replied:
Gotcha! I saw those feature flags being passed in but wasn't sure it was relevant here, but your explanation makes sense. I've pushed a change to remove these.

sla8c commented on WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift:
```diff
@@ -197,47 +182,46 @@ private extension SupportTableViewController {
                                                    action: supportEmailSelected(),
                                                    accessibilityIdentifier: "set-contact-email-button",
-                                                   featureFlagSupportForumEnabled: featureFlagStore.value(for: FeatureFlag.wordPressSupportForum)))
+                                                   featureFlagSupportForumEnabled: isForumShown))
```
@guarani parameter `featureFlagSupportForum` for class `SupportEmailRow` similar to my comment on `HelpRow`

guarani replied:
👍 Resolved alongside [similar changes](https://github.com/wordpress-mobile/WordPress-iOS/pull/20115#discussion_r1105940632).

Commit:
Removed now unused code

guarani commented:
I removed the review request while I check [this comment ](https://github.com/wordpress-mobile/WordPress-iOS/pull/20115#discussion_r1105890155) sorry for the noise.

Commit:
Remove now unused feature flag

sla8c reviewed (approved):
@guarani thanks LGTM!

guarani commented:
> @guarani thanks LGTM!

Thank you @sla8c!

Commit:
Merge branch 'trunk' into update/support-screen
